src/GeneratedMapper/MapperGenerator.cs

```diff
@@ -45,7 +46,7 @@ public void Execute(GeneratorExecutionContext context)
                 {
                     foreach (var (name, text) in GenerateMappings(information))
                     {
-                        context.AddSource(name, text);
+                        context.AddSource($"{name}.{++fileIndex}.g.cs", text);
```

Comment:
This seems like temp solution for unique, file naming.
Another option, which might be slightly less annoying is putting the Namespace in the FileName, that way it will always be unique but very long.
Less annoying is to not add sources but keep in a list, and add afterwards, and if any duplicate names then use namespaces, so that if there is no conflict you don't put namespaces, but do for the files that will conflict with each other.  Who knows if that has any memory impact whatsoever though.

This all begs the question how often will people go into these files.  Currently you have to search them out in solution explorer in order to do things like set breakpoints to debug things, so naming is kind of important that it's as readable as possible in these cases.
But other option is to catch Exceptions source gen throws and that will break inside generated code, so maybe file name isn't that big of a deal even now.

Reply:
I understand, but I wanted to keep the hint path to be small and concise, because otherwise the filename in the error list becomes unreadable. Also, finding the file in the solution explorer becomes quite hard, because you cannot double click on an error to go to the file when the file is generated. The whole experience with generated files is bad in VS, the files do not have any useful syntax highlighting or squiggly lines to highlight the error, and code navigation does not work either..

So I don't think people will go in these files normally, but only when they encounter an error. I don't see them keeping these files open for longer times, so the file index will probably remain the same when they are tweaking the mapping to fix a compile error.

Reply:
Will be fixed in next PR